core/client/fs/src/main/java/alluxio/client/file/FileInStream.java

```diff
@@ -98,8 +101,11 @@
    * stream reads from a remote worker.
    */
   protected BlockOutStream mCurrentCacheStream;
-  /** The blockId used in the block streams. */
-  private long mStreamBlockId;
+  /** Index of the ID of the current block stream in {@link #mBlockIds}. -1 means there is no
+   * current block stream yet. When the file stream has reached EOF, it is set to the length of
+   * {@link #mBlockIds}.
+   */
+  private int mBlockIdIndex = -1;
```

Comment:
can we define a private const `UNINITIALIZED_BLOCK_INDEX=-1` and use this constant for this init value of block index?

Reply:
done

Reply:
Same here, in order to better read the code, can we assign a constant value of maybe `EOF_BLOCK_INDEX=-2` to indicate EOF, rather than using the length of `#mBlockIds`

Reply:
is it true that, `mPos` is always the source of truth? e.g., we seem to always update `mBlockIdIndex` according to `mPos`, not vise versa.

Reply:
No, in `seek` or `skip`, `mBlockIndex` is updated according to `mPos`. But for streaming read, the streams are updated according to `mBlockIndex`.